waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -397,11 +420,13 @@ when defined(rln) or (not defined(rln) and not defined(rlnzerokit)):
     let verifyIsSuccessful = verify(rlnInstance, addr proofBuffer, addr f)
     if not verifyIsSuccessful:
       # something went wrong in verification
-      return false
+      return err("could not verify proof")
     # f = 0 means the proof is verified
-    if f == 0:
-      return true
-    return false
+    if f != 0:
+      return ok(false)
+
+    # Compare merkle tree roots for a valid proof
+    return validateVerifiedProof(rlnInstance, proof)
```

Comment:
We can save computation time if we check the root before proof verification. I'd move `validateVerifiedProof` before the call to the `verify` proc in line `420`. Also, the proc name `validateVerifiedProof` is a bit general while it only validates the root.

Reply:
Addressed in [fefb7e7](https://github.com/status-im/nwaku/pull/1158/commits/fefb7e706182ee964835d1393c5a9f7dc480ad9d)